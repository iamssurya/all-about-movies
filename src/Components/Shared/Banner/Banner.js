import React from "react";
import { Col, Container, Row } from "react-bootstrap";
import { MovieBannerBasePath } from "site-config";

import "./Banner.scss";

const mockImage = "http://image.tmdb.org/t/p/w1280/x4UkhIQuHIJyeeOTdcbZ3t3gBSa.jpg";

const Banner = (props) => {
  const {
    background,
    title,
    tagline,
    overview,
    releaseDate,
    hideDescription,
    runtime,
  } = props;
  const imageBackground = MovieBannerBasePath + background;
  return (
    <React.Fragment>
      <div
        className="banner"
        style={{
          background: `linear-gradient(to bottom, rgba(0,0,0,0)
          39%, rgba(0,0,0,0)
          41%, rgba(0,0,0,0.65)
          100%), url('${imageBackground || mockImage}'), #1c1c1c`,
          backgroundSize: "100%",
        }}
      >
        <Container className={{ hide: hideDescription }}>
          <Row>
            <Col className="movie-detail" xs={12} md={{ span: 6, offset: 6 }}>
              <h3>
                {title || "title"} <span>({releaseDate | "N/A"})</span>
              </h3>
              {runtime && <p>{runtime} mins</p>}
              {tagline && (
                <p>
                  <i>{tagline}</i>
                </p>
              )}
              <p>{overview || "overview"}</p>
            </Col>
          </Row>
        </Container>
      </div>
    </React.Fragment>
  );
};

export default React.memo(Banner);
