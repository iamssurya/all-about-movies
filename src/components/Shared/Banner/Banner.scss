@import "styles";

.banner {
  @include fade-in(.5s);
  padding-top: 15%;
  height: 500px;
}

.movie-detail {
  @include fade-in(.5s, .5s);

  background: $transparent-black;
  text-align: left;
  padding: 1%;
  min-height: 250px;

  .runtime {
    color: $dull-white;
  }

  .tagline {
    font-style: italic;
    color: $dull-white;
  }

  .overview {
    @include fade-in(1s, 1s);

    color: $gold;
  }
}
