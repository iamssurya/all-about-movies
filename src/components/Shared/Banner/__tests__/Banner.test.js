import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Banner from "../Banner";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe("<Banner /> component tests", () => {
  const props = {
    background: "test-bg.png",
    title: "title",
    tagline: "tagline",
    overview: "overview",
    releaseDate: "12-12-1990",
    runtime: 123,
  };

  let BannerComponent;
  let WrapperBannerComponent;
  let fetchedProps;
  beforeEach(() => {
    BannerComponent = mount(<Banner {...props} />);
    WrapperBannerComponent = toJson(BannerComponent);
    fetchedProps = WrapperBannerComponent.children[0].props;
  });

  it("Should render component correctly", () => {
    expect(WrapperBannerComponent).toMatchSnapshot();
  });
});
