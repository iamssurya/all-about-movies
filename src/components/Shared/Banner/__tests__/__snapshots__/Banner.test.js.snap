// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> component tests Should render component correctly 1`] = `
<Memo(Banner)
  background="test-bg.png"
  overview="overview"
  releaseDate="12-12-1990"
  runtime={123}
  tagline="tagline"
  title="title"
>
  <div
    className="banner"
    style={
      Object {
        "background": "url('http://image.tmdb.org/t/p/w1280test-bg.png'),#000000",
        "backgroundSize": "100%",
      }
    }
  >
    <Container
      fluid={false}
    >
      <div
        className="container"
      >
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col
              className="movie-detail"
              md={
                Object {
                  "offset": 6,
                  "span": 6,
                }
              }
              xs={12}
            >
              <div
                className="movie-detail col-md-6 col-12 offset-md-6"
              >
                <h3>
                  title
                   
                  <span>
                    (
                    0
                    )
                  </span>
                </h3>
                <p
                  className="runtime"
                >
                  123
                   mins
                </p>
                <p
                  className="tagline"
                >
                  tagline
                </p>
                <p
                  className="overview"
                >
                  overview
                </p>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </div>
</Memo(Banner)>
`;
